wikichron/controls_sidebar_decorator/CoEditingControlsSidebarDecorator.py

```diff
@@ -0,0 +1,277 @@
+
+from .BaseControlsSidebarDecorator import BaseControlsSidebarDecorator
+from lib.cytoscape_decorator.CoEditingStylesheet import CoEditingStylesheet
+from lib.networks.types.CoEditingNetwork import CoEditingNetwork
+import data_controller
+
+class CoEditingControlsSidebarDecorator(BaseControlsSidebarDecorator):
+
+    def __init__(self, sidebar):
+        super().__init__(sidebar)
+
+
+    def add_stats_section(self):
+        super().add_stats_section()
+        stats = CoEditingControlsSidebarDecorator.default_stats()
+        self.add_stats_content(stats)
+
+
+    @staticmethod
+    def default_stats(st1 = 'Nodes: ...', st2 = 'First User: ...', 
+            st3 = 'Edges: ...', st4 = 'Last User: ...', st5 = 'Communities: ...',
+            st6 = 'Max Hub Degree: ...'):
+
+        return [
+                html.Div([
+                    html.P(st1, className='left-element'),
+                    html.P(st2, className='right-element')
+                ]),
+                html.Div([
+                    html.P(st3, className='left-element'),
+                    html.P(st4, className='right-element')
+                ]),
+                html.Div([
+                    html.P(st5, className='left-element', id='n_communities'),
+                    html.P(st6, className='right-element')
+                ])]
+
+
+    def add_metrics_section(self):
+        super().add_metrics_section()
+
+        metrics = [
+                html.Div(children=[
+                    html.Span('PageRank', className='left-element'),
+                    html.Button('Run', id='calculate_page_rank',
+                                type='button',
+                                className='right-element action-button'),
+                    ],
+                    className='metrics-section'),
+                html.Div([
+                    html.Span('Communities', className='left-element'),
+                    html.Button('Run', id='calculate_communities',
+                                type='button',
+                                className='right-element action-button')
+                ],
+                className='metrics-section')
+            ]
+
+        self.add_metrics_content(metrics)
+
+
+    def add_options_section(self):
+        super().add_options_section()
+
+        options = [
+                    html.Button('Show Labels', id='show_labels',
+                        className='control-button action-button'),
+                    html.Button('Show PageRank', id='show_page_rank',
+                        className='control-button action-button'),
+                    html.Button('Color by Cluster', id='color_cluster',
+                        className='control-button action-button'),
+                ]
+
+        self.add_options_content(options)
+
+
+    def add_all_sections(self):
+        self.add_stats_section()
+        self.add_metrics_section()
+        self.add_options_section()
+
+
+def bind_callbacks(app):
```

Comment:
We said that this could be a [@classmethod](https://docs.python.org/3/library/functions.html#classmethod).
Nevertheless, I don't see any specific reason to do so right now, so I'm fine with leaving it like this.

Reply:
In a second thought, I think bind_callbacks should be a @classmethod inside `BaseControlsSidebarDecorator`, don't you think so?

Reply:
I'm with u, the inherit classes have to implement it

Reply:
OK, do this and we're good to go :horse_racing:

Reply:
Done in https://github.com/Grasia/WikiChron-networks/commit/ff4e8f447c0010a8ca16560fe2c47c530ad9ac58